platform/mbed_retarget.h

```diff
@@ -619,6 +624,31 @@ std::FILE *fdopen(mbed::FileHandle *fh, const char *mode);
  */
 int bind_to_fd(mbed::FileHandle *fh);
 
+#else
+/** Targets may implement this to override how to write to the console.
+ *
+ * If the target hasn't provided minimal_console_putc, this is called
+ * to give the target a chance to specify a serial access to the console.
+ *
+ * If this is not provided, the console will be MinimalConsole.
```

Comment:
`MinimalConsole` is not public API, so this isn't meaningful.

Should say what the default behaviour actually is - to use `serial_putc` if `target.console-uart` is true, else no output.